### PR TITLE
Docker and cors

### PR DESCRIPTION
**Pull request by uthopiko**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any
| License       | MIT
| Doc PR        | reference to the documentation PR, if any

### CODE REVIEW

File: etc/bash/docker.sh [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
-docker-compose up -d
+cp .env.dist .env || echo "Previosuly created env"
+docker-compose up -d || docker-compose up -d
 docker-compose exec php bash -c "cd /var/www/taskmanager && composer install"
 docker-compose exec php bash -c "cd /var/www/identityaccess && composer install"
+docker-compose exec php bash -c "cd /var/www/identityaccess/var && chown www-data:www-data * && rm -rf cache/*"
+docker-compose exec php bash -c "cd /var/www/taskmanager/var && chown www-data:www-data * && rm -rf cache/*"
+docker-compose exec php bash -c "cd /var/www/identityaccess && sh etc/bash/generate_ssh_keys.sh"
+docker-compose exec php bash -c "echo '172.18.0.6 identityaccess.localhost' >> /etc/hosts"
+docker-compose exec php bash -c "echo '172.18.0.6 taskmanager.localhost' >> /etc/hosts"
```
Comment (gorkalaucirica) [inline etc/bash/docker.sh]: Not sure if it is always the same IP

File: etc/bash/docker.sh [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+
```
Comment (gorkalaucirica) [inline etc/bash/docker.sh]: Not sure if it is always the same IP, maybe we can get server's current IP and set it there

File: etc/docker/.env.dist [hunk 3]
```diff
@@ -12,10 +12,16 @@ MYSQL_ROOT_PASSWORD=root
 MYSQL_DATABASE=kreta
 MYSQL_USER=kreta
 MYSQL_PASSWORD=kreta
+MYSQL_EXTERNAL_PORT=3307
+MYSQL_INTERNAL_PORT=3307
```
Comment (gorkalaucirica) [inline etc/docker/.env.dist]: I think internal should be 3306 as set in parameters.yml

File: etc/docker/docker-compose.yml [hunk 4]
```diff
@@ -72,4 +72,4 @@ services:
         volumes_from:
             - php
         volumes:
-            - ./data/logs/nginx/:/var/log/nginx
+            - ./data/logs/nginx/:/var/log/nginx
```
Comment (gorkalaucirica) [inline etc/docker/docker-compose.yml]: Line missing

File: CompositeUi/src/Config.js [hunk 5]
```diff
@@ -21,8 +21,8 @@ let Config = {
 };
 
 Config = Object.assign({
-  taskManagerUrl: `//${Config.baseUrl}:8001`,
-  identityAccessUrl: `//${Config.baseUrl}:8002`,
+  taskManagerUrl: `//taskmanager.localhost`,
+  identityAccessUrl: `//identityaccess.localhost`,
```
Comment (gorkalaucirica) [inline CompositeUi/src/Config.js]: We need to make this two addresses configurable, maybe in a next PR

File: CompositeUi/src/scss/components/_notification.scss [hunk 6]
```diff
@@ -53,5 +53,5 @@ $notification-height: 50px !default;
   height: 15px;
   margin-left: 10px;
   margin-right: 10px;
-  width: 15px
+  width: 15px;
```
Comment (gorkalaucirica) [inline CompositeUi/src/scss/components/_notification.scss]: 🎉

Comment (benatespina) [inline CompositeUi/src/scss/components/_notification.scss]: This bug is already fixed in the master branch. You need to rebase master before merge this PR.

File: etc/docker/.env.dist [hunk 7]
```diff
@@ -12,10 +12,16 @@ MYSQL_ROOT_PASSWORD=root
...
 RABBITMQ_DEFAULT_PASS=kreta
+RABBITMQ_EXTERNAL_PORT=5673
+RABBITMQ_INTERNAL_PORT=5672
+RABBITMQ_EXTERNAL_GUI_PORT=15673
+RABBITMQ_INTERNAL_GUI_PORT=15672
 
 TASK_MANAGER_HOST=taskmanager.localhost
 IDENTITY_ACCESS_HOST=identityaccess.localhost
-COMPOSITE_UI_HOST=kreta.localhost
+COMPOSITE_UI_HOST=kreta.localhost
```
Comment (benatespina) [inline etc/docker/.env.dist]: EOF missing

File: etc/docker/docker-compose.yml (hunk 4 above)
Comment (uthopiko) [inline etc/docker/docker-compose.yml]: I don't understand this...

File: etc/bash/docker.sh [hunk 8]
```diff
@@ -1,6 +1,26 @@
 #!/usr/bin/env bash
 
 cd $(dirname $0)/../docker
-docker-compose up -d
-docker-compose exec php bash -c "cd /var/www/taskmanager && composer install"
-docker-compose exec php bash -c "cd /var/www/identityaccess && composer install"
+cp .env.dist .env || echo "Previosuly created env"
```
Comment (gorkalaucirica) [inline etc/bash/docker.sh]: Any possibility to check that .env exists before copying? Right now the `.env` is overwritten even if exists

File: etc/docker/docker-compose.yml [hunk 9]
```diff
@@ -73,3 +73,13 @@ services:
...
+        networks:
+            default:
+              ipv4_address: 172.18.0.10
+
+networks:
+  default:
+    driver: bridge
+    ipam:
+      config:
+      - subnet: 172.18.0.0/24
```
Comment (gorkalaucirica) [inline etc/docker/docker-compose.yml]: Line missing

File: etc/docker/README.md [hunk 10]
```diff
@@ -11,57 +11,23 @@ It provides the following containers:
...
 
 
-2. Build/run containers with (with and without detached mode)
+2. Update your system host file
 
+    Add this line to /etc/hosts 
     ```bash
-    $ docker-compose build
-    $ docker-compose up -d
+    "127.0.0.1 kreta.localhost taskmanager.localhost identityaccess.localhost"
```
Comment (gorkalaucirica) [inline etc/docker/README.md]: Quotes are not needed

Comment (uthopiko) [inline etc/docker/README.md]: And in troubleshooting section add 502 unexpected behaviour explanation until we get a better choice for run node

**Pull request merged**